[WIP] Message Browser

tesonep opened the pull request:
Making the Message browser in a good shape

- Requires https://github.com/pharo-project/pharo/pull/18082
- Then it allows us to merge https://github.com/pharo-project/pharo/pull/18085
- There are some issues open that should be nice before integrating this or at least before integrating the Pharo PR:
   - #1086 
   - #1087 
   - #1088 
   - #1090
   - #1097
   - #1100

tesonep committed:
- Adding test package for StMessageBrowser
- Adding tests
- Fixing it to open
- Fixing Title Issue

Ducasse commented:
Thanks pablo your tests are inspiring for me. 
I will check if I can write tests for the object transcript in a similar vein.

tesonep committed:
- Adding tests to cover case
- Open senders and implementors mantain the selected scope

tesonep committed:
- Better handling of scopes
- Adding order for the list
- Removing duplications in the scope
- Implementing missing functionality to handle Calypso integration

tesonep committed:
- Organizing Tests
- Adding tests for updates of the list
- Fixing To show comments
- Updating Scopes

guillep committed:
Fix tests

guillep committed:
Fix autoselection/highlight (see https://github.com/pharo-project/pharo/issues/18301)

guillep commented:
If you load this in a new image, be careful to execute before: `Smalltalk tools initDefaultToolSet`

guillep committed:
Messages should be a growable collection

Ducasse commented:
@tesonep I will close this issue since message browser got a lot of effort.
